src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/SaveToMinioTasklet.java

```diff
@@ -1,22 +1,78 @@
+import static org.folio.dew.utils.Constants.getWorkingDirectory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.folio.dew.batch.ExecutionContextUtils;
+import org.folio.dew.batch.acquisitions.edifact.services.OrganizationsService;
+import org.folio.dew.domain.dto.VendorEdiOrdersExportConfig;
+import org.folio.dew.repository.LocalFilesStorage;
+import org.folio.dew.repository.RemoteFilesStorage;
 import org.springframework.batch.core.StepContribution;
 import org.springframework.batch.core.configuration.annotation.StepScope;
 import org.springframework.batch.core.scope.context.ChunkContext;
 import org.springframework.batch.core.step.tasklet.Tasklet;
 import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
 import org.springframework.stereotype.Component;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
 import lombok.extern.log4j.Log4j2;
 
 @RequiredArgsConstructor
 @Component
 @StepScope
 @Log4j2
 public class SaveToMinioTasklet implements Tasklet {
-  @Override public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) {
-    // TODO: implement minio upload
+  private final RemoteFilesStorage remoteFilesStorage;
+  private final LocalFilesStorage localFilesStorage;
+  private final OrganizationsService organizationsService;
+  private final ObjectMapper objectMapper;
+
+  @Value("${spring.application.name}")
+  protected String springApplicationName;
+  @Value("#{jobParameters['edifactOrderAsString']}")
+  private String edifactOrderAsString;
+
+  @Override
+  @SneakyThrows
+  public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) {
+    // retrieve parameters from job context
+    var jobParameters = chunkContext.getStepContext().getJobParameters();
+    var ediExportConfig = objectMapper.readValue((String)jobParameters.get("edifactOrdersExport"), VendorEdiOrdersExportConfig.class);
+
+    var downloadFilename = createTempFile(ediExportConfig, edifactOrderAsString);
+
+    var uploadedFilePath = remoteFilesStorage.uploadObject(edifactOrderAsString, downloadFilename, downloadFilename, MediaType.TEXT_PLAIN_VALUE, false);
+    ExecutionContextUtils.addToJobExecutionContext(contribution.getStepExecution(), "uploadedFilePath", uploadedFilePath, "");
+
     return RepeatStatus.FINISHED;
   }
+
+  private String generateFileName(VendorEdiOrdersExportConfig ediExportConfig) {
+    var vendorId = ediExportConfig.getVendorId().toString();
+    var vendor = organizationsService.getOrganizationById(vendorId);
+    var vendorName = vendor.get("code").asText();
+    DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss");
```

Comment:
nitpick: dateFormat can be declared as a constant/final property at the top of the file.

Reply:
fixed